### PR TITLE
Fix compiler issues on clang 14 + FreeBSD

### PR DESCRIPTION
**Pull request by erikarn**
clang complains about duplicate symbols for some global symbols and enums. It's, well, technically correct.  There are global variables and enums being defined in .h files that are really multiple definitions of the same thing when included by multiple things.

So, well, technically speaking, LLVM's ld was complaining about it.

So, correct them:

* global symbols are now externs in the header files and the variables are now defined in .c source files.
* enums are now typdef enums, so they're not effectively defining variables with those names in them.

Tested - compiled/run on FreeBSD 14

### CODE REVIEW

**Comment by rofl0r** [comment]
it appears clang now defaults to -fno-common, like gcc 10+. i'm fine with the extern decls of the variables, but i'm not seeing what problem the enum typedefs are trying to solve.
can you show the corresponding compiler errors?

**Comment by erikarn** [comment]
It's throwing linker errors. Those enums are defined wrong:

enum {

} foo;

defines an enum and a variable 'foo'.

enum foo {

};

defines an enum named 'foo'.

typedef enum {

} foo;

creates a type 'foo' that's an enum.

Another option is to move the 'foo' from the bottom (where it's being treated as a variable) to the top (where it defines an 'enum foo').

**Comment by rofl0r** [comment]
> Another option is to move the 'foo' from the bottom (where it's being treated as a variable) to the top (where it defines an 'enum foo').

that sounds like it was the original intention, so it would be the preferable solution. it's also clearer when you see a function signature accepting enum foo instead of something typedeffed - it's immediately clear we're dealing with an enum.

**Comment by erikarn** [comment]
ok! I just pushed an update that changes the enums to that!



-a

File: src/cpc/crtc.h [hunk 1]
```diff
@@ -208,7 +208,7 @@ void	CRTC_SetRenderState(BOOL State);
 
 void	CRTC_SetTrueColourRender(BOOL State);
 
-enum
+typedef enum
 {
```
Comment (rofl0r) [inline src/cpc/crtc.h:212]: seems you overlooked this one ?

Comment (erikarn) [reply src/cpc/crtc.h:212]: yup you're right and i did miss it! I've fixed that now, would you mind taking another look?

Thanks!

**Comment by erikarn** [comment]
pushed, sorry it took so long, please re-review!

**Pull request merged**